### ISSUE TITLE
suggestion to add an ability to hide apis and libraries like in modmenu

### ISSUE THREAD

**Issue by delveticas** [MrCrayfish/Catalogue#60]

**Comment by Tapacywka** [comment]
Yes, please, we need this feature for our modpack, especially with Sinytra Connector where 100 additional libraries are presented.

**Comment by mpustovoi** [comment]
It's too bad that this important feature isn't being paid attention at all

**Comment by Boshigiri** [comment]
Yes, Sinytra Connector add lots of libraries.
We need this feature.

**Comment by MrCrayfish** [comment]
@Boshigiri @Tapacywka Unfortunately a lot of authors don't mark their mod as a library in NeoForge (using `FMLModType` in the JAR manifest), so it's impossible for Catalogue to determine if a mod is actually a library. Nested jars should be marked as a library in NeoForge, however Sinytra might not be following the same behaviour.

**Comment by Boshigiri** [comment]
Because i tried the neoforge ported Modmenu(Better Modlist [(Neo)Forge]), it can hide the fabric api correctly, I thought it could be detected
So i should submit this issue to Sinytra Team?

**Comment by MrCrayfish** [comment]
I've just reported an issue outlining the problem (https://github.com/Sinytra/Connector/issues/1524). I can mark Fabric API libraries correctly, but Jar in Jar libraries from regular Fabric mods won't be marked as libraries based on the code I've read in Sinytra Connector.

**Comment by MrCrayfish** [comment]
I've made a patch that will be available in 1.21.1
https://github.com/MrCrayfish/Catalogue/commit/9902144d9031eb5edaaf8b6e49a91f021fdb0d85

**Comment by Boshigiri** [comment]
Thank you for your work, just being able to hide the fabric api would be useful enough

**Comment by MrCrayfish** [comment]
I've also made it possible to hide libraries in the list, assuming they are also marked as libraries. Added in https://github.com/MrCrayfish/Catalogue/commit/15265572a4af1c5b042c3d3b31eaa5e4325bb06f

**Comment by Furglitch** [comment]
I was going to make a separate feature request, but seeing as this wasn't closed with the commit, I figured it would be good to put here and not add to the number of issues. I can make a separate issue for the request if deemed necessary.

As you mentioned, the main problem with the libraries feature is that many devs don't properly

Perhaps some kind of config file could be made? Like, a line-by-line list of libraries that could be maintained by end-users both on their local instances and through pull requests to keep track of libraries? Catalogue already provides mod IDs, so the list could utilize those. Something like...

```plaintext
almanac
balm
bookshelf
cloth_config
creative core
```

**Comment by staterral** [comment]
Hi, sorry for reviving a few months long dead issue, but I am very interested in Furglitch's idea of adding a feature to manually hide mods in the config file/menu.

Would you be interested in exploring this idea in the future? it would be really helpful for modpack makers. Thanks
